Sources/XcodeProj/Objects/Configuration/BuildSettings.swift

```diff
@@ -1,4 +1,69 @@
 import Foundation
 
 /// Build settings.
-public typealias BuildSettings = [String: Any]
+public typealias BuildSettings = [String: BuildSetting]
+
+public enum BuildSetting: Sendable, Equatable {
+    case string(String)
+    case array([String])
+
+    var valueForWriting: String {
+        switch self {
+        case let .string(string):
+            string
+        case let .array(array):
+            array.joined(separator: " ")
+        }
+    }
+
+    public var stringValue: String? {
+        if case let .string(value) = self {
+            value
+        } else {
+            nil
+        }
+    }
+
+    public var arrayValue: [String]? {
```

Comment:
would probably also remove this in favor of consumers doing the switch themselves.